### PR TITLE
dcp-491 download spreadsheet timeout

### PR DESCRIPTION
**Pull request by prabh-t**
ebi-ait/dcp-ingest-central#491

- user notice on screen
- added timeout of *10* mins (may need to increase) for spreadsheet download request
- disable download button for a minimum time (20 mins) when download request is timed out - to avoid successive download requests which may overload the cluster for large submissions.

### CODE REVIEW

**Review by amnonkhen — COMMENTED** [review]
Looks OK, but lacks testing code. @prabh-t , please let me know if you need help adding tests.

File: src/app/shared/services/broker.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+          expect(err.statusText).toEqual('Request Timeout');
+          done();
+        });
+
+      let req = httpTestingController.expectOne(`${api_url}/submissions/${submissionUuid}/spreadsheet`);
+      req.flush(null, {
+        status: 408,
+        statusText: 'Request Timeout'
+      });
+    });
```
Comment (aaclan-ebi) [inline src/app/shared/services/broker.service.spec.ts:71]: I don't think this is testing the timeout error if no value is emitted before DOWNLOAD_SPREADSHEET_TIMEOUT. 

You can try delay to delay the response:
```
of(response).delay(DOWNLOAD_SPREADSHEET_TIMEOUT + 5 secs);
   }
```
and tick to fast forward to timeout time 
```
tick(DOWNLOAD_SPREADSHEET_TIMEOUT); // fast forward to timeout time
```

Comment (aaclan-ebi) [reply src/app/shared/services/broker.service.spec.ts:71]: or  
```
setTimeout(() => {
 req.flush(null, {
        status: 408,
        statusText: 'Request Timeout'
      });
  }, DOWNLOAD_SPREADSHEET_TIMEOUT + 1*60*1000);

**Review by aaclan-ebi — APPROVED** [review]

File: src/app/shared/services/broker.service.spec.ts (hunk 1 above)
Comment (aaclan-ebi) [reply src/app/shared/services/broker.service.spec.ts]: nvm there's no need to set timeout for the response

**Review by aaclan-ebi — APPROVED** [review]

**Pull request merged**